src/tools/run-make-support/src/rustc.rs

```diff
@@ -150,6 +150,19 @@ impl Rustc {
         self
     }
 
+    /// Favours dynamic linking over static linking in rustc. Equivalent to `-C prefer-dynamic``.
+    pub fn prefer_dynamic(&mut self) -> &mut Self {
```

Comment:
Yeah, as Nils said, I think we probably just use -Cprefer-dynamic directly, since the helper doesn't add much